akka-persistence/src/test/scala/akka/persistence/PersistentActorStashingSpec.scala

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.persistence
+
+import akka.actor.ActorRef
+import akka.persistence.journal.SteppingInmemJournal
+import akka.testkit.ImplicitSender
+import com.typesafe.config.Config
+
+import scala.concurrent.duration._
+
+object PersistentActorStashingSpec {
+  final case class Cmd(data: Any)
+  final case class Evt(data: Any)
+
+  abstract class ExamplePersistentActor(name: String) extends NamedPersistentActor(name) {
+    var events: List[Any] = Nil
+    var askedForDelete: Option[ActorRef] = None
+
+    val updateState: Receive = {
+      case Evt(data)               ⇒ events = data :: events
+      case d @ Some(ref: ActorRef) ⇒ askedForDelete = d.asInstanceOf[Some[ActorRef]]
+    }
+
+    val commonBehavior: Receive = {
+      case "boom"   ⇒ throw new TestException("boom")
+      case GetState ⇒ sender() ! events.reverse
+    }
+
+    def receiveRecover = updateState
+  }
+
+  class UserStashPersistentActor(name: String) extends ExamplePersistentActor(name) {
+    var stashed = false
+    val receiveCommand: Receive = {
+      case Cmd("a") ⇒ if (!stashed) { stash(); stashed = true } else sender() ! "a"
+      case Cmd("b") ⇒ persist(Evt("b"))(evt ⇒ sender() ! evt.data)
+      case Cmd("c") ⇒ unstashAll(); sender() ! "c"
+    }
+  }
+
+  class UserStashManyPersistentActor(name: String) extends ExamplePersistentActor(name) {
+    val receiveCommand: Receive = commonBehavior orElse {
+      case Cmd("a") ⇒ persist(Evt("a")) { evt ⇒
+        updateState(evt)
+        context.become(processC)
+      }
+      case Cmd("b-1") ⇒ persist(Evt("b-1"))(updateState)
+      case Cmd("b-2") ⇒ persist(Evt("b-2"))(updateState)
+    }
+
+    val processC: Receive = {
+      case Cmd("c") ⇒
+        persist(Evt("c")) { evt ⇒
+          updateState(evt)
+          context.unbecome()
+        }
+        unstashAll()
+      case other ⇒ stash()
+    }
+  }
+
+  class UserStashFailurePersistentActor(name: String) extends ExamplePersistentActor(name) {
+    val receiveCommand: Receive = commonBehavior orElse {
+      case Cmd(data) ⇒
+        if (data == "b-2") throw new TestException("boom")
+        persist(Evt(data)) { event ⇒
+          updateState(event)
+          if (data == "a") context.become(otherCommandHandler)
+        }
+    }
+
+    val otherCommandHandler: Receive = {
+      case Cmd("c") ⇒
+        persist(Evt("c")) { event ⇒
+          updateState(event)
+          context.unbecome()
+        }
+        unstashAll()
+      case other ⇒ stash()
+    }
+  }
+
+  class AsyncStashingPersistentActor(name: String) extends ExamplePersistentActor(name) {
+    var stashed = false
+    val receiveCommand: Receive = commonBehavior orElse {
+      case Cmd("a") ⇒ persistAsync(Evt("a"))(updateState)
+      case Cmd("b") if !stashed ⇒
```

Comment:
Oh interesting... scalariform must be freaking out because the code block seems like multi line maybe... due to the `;` or something?

Don't worry about it then :)